src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/JsonRPCFaultExceptionHandler.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Razor.Telemetry;
+using StreamJsonRpc;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer;
+
+internal class JsonRPCFaultExceptionHandler : IFaultExceptionHandler
+{
+    public bool HandleException(ITelemetryReporter reporter, Exception exception, string? message, object?[] @params)
+    {
+        if (exception is not RemoteInvocationException  remoteInvocationException)
+        {
+            return false;
+        }
+
+        reporter.ReportFault(remoteInvocationException, remoteInvocationException.Message, remoteInvocationException.ErrorCode, remoteInvocationException.ErrorData, remoteInvocationException.DeserializedErrorData);
```

Comment:
Does the actual exception give us _no_ useful information at all? I would have thought we'd somehow want to combine all of the info together.

Reply:
It does, and we're still keeping that info. We're just adding more now